src/helpers.tsx

```diff
@@ -1,29 +1,50 @@
 /* eslint-disable import/prefer-default-export */
+import { logger } from '@storybook/client-logger';
 import { WrapperSettings } from '@storybook/addons';
 import { DEFAULT_PADDING } from './constants';
 
-type Option = {
-  name: string;
-  value: string;
-  default?: boolean;
+type Value = { name: string; value: string };
+
+type Values = { [key: string]: Value };
+
+type PaddingConfig = {
+  default: string;
+  disable?: boolean;
+  values: Values;
 }
 
-type Options = Option[] | WrapperSettings['parameters'];
+type Option = Value & {default?: boolean}
+
+type Options = Option[] | PaddingConfig | WrapperSettings['parameters'];
+
+export const normalizeValues = (options: Options) => (Array.isArray(options)
+  ? options
+  : Object.entries<Value>(options.values).map(([key, { name, value }]) => {
+    const isDefault = options.default === key;
 
-export const isEnabled = (options: Options): options is Option[] => options.length > 0;
+    return { name, value, default: isDefault };
+  }));
 
-export const getSelectedPadding = (options: Options, currentValue: string): string => {
-  if (!isEnabled(options)) {
-    return DEFAULT_PADDING;
+export const isEnabled = (options: Options) => {
+  if (Array.isArray(options)) {
+    logger.warn(
+      'Storybook addon paddings has changed, please migrate to the new recommended configuration.',
```

Comment:
Please, let me know your thoughts on the message 😬

Reply:
I would suggest something like:

> Using array parameter with Paddings addon is deprecated, and support will be dropped in a future major version. Please refer to the release notes for migration: https://github.com/rbardini/storybook-addon-paddings/releases/tag/v2.1.0

(assuming v2.1.0 will be the next release)